x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/CohereService.java

```diff
@@ -78,16 +79,36 @@ public CohereModel parseRequestConfig(
         return model;
     }
 
-    private static CohereModel createModel(
+    private static CohereModel createModelWithoutLoggingDeprecations(
```

Comment:
We only want to print deprecation log messages when a rest request is made. Otherwise if the `model` field is used in the inference entity we'd print the message every time we load it which wouldn't be ideal.